keys_storage.go

```diff
@@ -1,19 +1,32 @@
 package doubleratchet
 
+import (
+	"bytes"
+	"sort"
+)
+
+type SessionID = []byte
+
 // KeysStorage is an interface of an abstract in-memory or persistent keys storage.
 type KeysStorage interface {
 	// Get returns a message key by the given key and message number.
 	Get(k Key, msgNum uint) (mk Key, ok bool, err error)
 
 	// Put saves the given mk under the specified key and msgNum.
-	Put(k Key, msgNum uint, mk Key) error
+	Put(sessionID SessionID, k Key, msgNum uint, mk Key, keySeqNum uint) error
```

Comment:
The args list becomes quite big, and usually, it's a sign to group those args into a separate struct. Seems like  `InMemoryKey` type already has all the same fields. What do you think?

Reply:
`InMemoryKey` is only used by the toy in memory implementation that we provide, but we could have a struct passed, I would personally hold on for now as I'd like to validate the changes with some usage, to make sure the implementation is sound, and then we get back and refactor, what do you reckon?